cli/azd/pkg/azdo/repository.go

```diff
@@ -101,10 +101,15 @@ func GetGitRepositoriesInProject(
 	}
 	repos := *getRepositoriesResult
 
-	options := make([]string, len(repos))
-	for idx, repo := range repos {
-		options[idx] = *repo.Name
+	options := make([]string, 0, len(repos))
+	for _, repo := range repos {
+		options = append(options, *repo.Name)
+	}
+
+	if len(options) == 0 {
+		return nil, fmt.Errorf("no repositories found in project %s", projectName)
 	}
```

Comment:
One additional protection we could add would be to modify our wrapper of `console.Select(...)` to detect these cases where the options have a length of zero and return an error vs the prompt with an empty list.

Reply:
This is indeed the current behavior. It returns an error `please provide options to select from`, not "prompt with an empty list".